### PR TITLE
Fixes #12707 Nest climate card contains redundant fields

### PR DESCRIPTION
**Pull request by ciotlosm**
Fixes https://github.com/home-assistant/home-assistant/issues/12707 Nest climate card contains redundant fields

### CODE REVIEW

File: src/dialogs/more-info/controls/more-info-climate.js [hunk 1]
```diff
@@ -309,11 +309,11 @@ class MoreInfoClimate extends EventsMixin(PolymerElement) {
   }
 
   supportsTemperature(stateObj) {
-    return (stateObj.attributes.supported_features & 1) !== 0;
+    return ((stateObj.attributes.supported_features & 1) !== 0) && stateObj.attributes.temperature;
```
Comment (c727) [inline src/dialogs/more-info/controls/more-info-climate.js]: temperature can be 0 (= false), some for other attributes

File: src/dialogs/more-info/controls/more-info-climate.js [hunk 2]
```diff
@@ -309,11 +309,11 @@ class MoreInfoClimate extends EventsMixin(PolymerElement) {
...
   }
 
   supportsTemperatureRange(stateObj) {
-    return (stateObj.attributes.supported_features & 6) !== 0;
+    return ((stateObj.attributes.supported_features & 6) !== 0) && (stateObj.attributes.target_temp_low || stateObj.attributes.target_temp_high);
```
Comment (balloob) [inline src/dialogs/more-info/controls/more-info-climate.js]: test with `('target_temp_low' in stateObj.attributes)`

File: src/dialogs/more-info/controls/more-info-climate.js (hunk 1 above)
Comment (ciotlosm) [reply src/dialogs/more-info/controls/more-info-climate.js]: That is what you get for reusing previous code :-) I will fix it by using @balloob suggestion

File: src/dialogs/more-info/controls/more-info-climate.js (hunk 2 above)
Comment (ciotlosm) [reply src/dialogs/more-info/controls/more-info-climate.js]: Unfortunately this doesn't really work for the intended need:
screen shot 2018-05-31 at 00 17 25

**Review by c727 — APPROVED** [review]

**Pull request merged**